### PR TITLE
Performing Lighthouse SEO recommendations

### PR DESCRIPTION
**Pull request by MikeRogers0**
I searched for LRUG on DDG & we [weren't ranking very well](https://user-images.githubusercontent.com/325384/93709689-32f6d900-fb38-11ea-87d6-4e0add53b8bd.png). So I had a little investigation into what could be causing that :)

Our [lighthouse report](https://lighthouse-dot-webdotdevsite.appspot.com//lh/html?url=https%3A%2F%2Flrug.org) showed we had some aspects we could improve upon:

image

I also ran the site through a [HTML Validator](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Flrug.org%2F) & noted we had room for improvement also.

# What this changes

Mostly I just went through and made the "quick win" changes, such as improving HTML validation & adding suitable metatags.

### CODE REVIEW

File: source/_footer.erb
Comment (MikeRogers0) [inline source/_footer.erb:2]: image

The partial render markdown, which lead to a `<p>` within a `<p>`. 

I decided the right solution there was to just adjust the font-size in the footer to be the same as small.

File: source/_header.erb
Comment (MikeRogers0) [inline source/_header.erb:5]: image

The headings are in a bit of a funny order, but no `<h1>` isn't ideal. So I added one :)

image

I kind of liked the slightly bigger font size & heaver weight, what do you think?

File: source/_sponsors.erb
Comment (MikeRogers0) [inline source/_sponsors.erb:3]: image

image


The `<h4>` was kind of tricky on this one, but I decided an `<aside>` tag should make the semantics a bit more improved.

File: source/index.html.erb
Comment (MikeRogers0) [inline source/index.html.erb:14]: image

I wanted the homepage to have a description which uses all our keywords over being blank, this worked nicely.

File: source/index.html.erb
Comment (MikeRogers0) [inline source/index.html.erb:21]: Because Kramdown generates IDs for headers (which is awesome), it would create duplicate ids on the homepage. I added a way to pragmatically skip the generation of the header IDs based on context.

File: source/layouts/base.erb
Comment (MikeRogers0) [inline source/layouts/base.erb:2]: image

File: source/layouts/base.erb
Comment (MikeRogers0) [inline source/layouts/base.erb:5]: Our meta open graph links were using `http`, so this should speed up the initial time it takes to open a link to the page.

File: source/layouts/base.erb
Comment (MikeRogers0) [inline source/layouts/base.erb:14]: image

File: source/_header.erb
Comment (h-lame) [reply source/_header.erb:5]: Yeah, it's nice.  No problem with making this an `<h1>` visually at least.  However, I think there would be a problem on pages other than the main index page as we'd end up with 2 `<h1>`s.  The main index page doesn't have a page title - we just go straight into the meetings index and render via `source/layouts/base.erb` which has no `<h1>`.  However, if we look at an individual meeting page, those go through `source/layouts/layout.erb` which _does_ have a page title `<h1>` so if we make the thing in the title an `<h1>` we'd have two on the page, and that's as bad as not having one, right?

But ...

As I look at this, and the markup around it, I wonder if the `<div id="header">` wrapper should just be be `<header>`?  That should let us have two `<h1>` elements on a page rendered via `source/layouts/layout.erb`. Which brings us back to the main index not having an `<h1>`.  I think we could make the `<div>` from `source/meetings/_meeting.erb` into an `<article>` and then make the existing `<h2>` for the meeting title into an `<h1>`.  Dunno what the semantics are of not having a "page-scoped" `<h1>` on a page with lots of `<article>`s though.  Maybe it's _as_ bad as having none, or having two.

Aside (pun intended): If we did any of this we should probably make `<div id="footer">` into a `<footer>` too, while we're at it.  Not sure how far we want to take this - the markup is pretty mid-late 2000s and `<div>`-heavy and we might disappear down a rabbit hole.

**Review by h-lame — COMMENTED** [review]
This is great, thanks for doing it.  Never even occurred to me to push the site through any SEO tools.  It doesn't surprise me _at all_ that there are simple semantic issues.  Left some comments though that it'd be nice to have some back-and-forth on, before I'd be ready to merge.

File: source/_sponsors.erb
Comment (MikeRogers0) [reply source/_sponsors.erb]: It looks like `<h1>` is the right choice when in an aside https://www.w3.org/TR/2014/REC-html5-20141028/sections.html#the-aside-element :D

File: source/_header.erb
Comment (MikeRogers0) [reply source/_header.erb:5]: I added `<header>`/`<footer>`/`<article>` tags around the various elements, and Lighthouse seemed pretty happy with that approach on both the homepage & subpages.

File: lib/lrug_helpers.rb
Comment (MikeRogers0) [inline lib/lrug_helpers.rb:134]: I added native lazy loading, it appears to drop our homepage loading time quite nicely so I've kept it in.

**Comment by MikeRogers0** [comment]
I've updated this to include @h-lame's requested changes.

I think we've improve the semantics of the pages a lot more, plus added a few small tweaks to improve the page performance.

File: source/_footer.erb
Comment (h-lame) [inline source/_footer.erb:1]: Should we remove the redundant `id`s too?  I can't imagine needing a 2nd `<footer>` element so it should be safe to rewrite any CSS selectors from `#footer { ... }` to `footer { ... }` right?  Same goes for `header`.

I'm wary of pushing too much into one PR though and exhausting your patience.  So maybe this is one for a future PR - there's a _lot_ of `id` based styling in here that could probably be class or element based.

File: source/meetings/_meeting.erb
Comment (h-lame) [inline source/meetings/_meeting.erb:2]: This could be an `<h1>` as well I think.

File: source/meetings/_meeting.erb
Comment (h-lame) [inline source/meetings/_meeting.erb:4]: Oh, neat - glad this can just be passed through! 👏

**Review by h-lame — COMMENTED** [review]
This looks great. 

Left a few comments but mostly they're curiosity or idle thoughts that won't hold up this PR.  I would like to address the `<h2>` -> `<h1>` in `source/meetings/_meeting.erb` though.

Other than that the only other question I have is where the heading-level-outline screenshots came from?  Perhaps I'm just not clicking through all the links in the lighthouse report but I can't find them.  Be useful to know how to generate them ourselves to check these kinds of things.

File: source/stylesheets/components/_sponsors.scss
Comment (MikeRogers0) [reply source/stylesheets/components/_sponsors.scss:23]: Currently they look like:

image

I'm guessing that element has changed at some point, but setting CSS like above overrides the explicit HTML attributes to make them fit a bit more nicely.

I'm 50/50 on if this was relevant to this PR, but as I was touching that file already it took my focus.

The end result is it looks more like:

image

Plus also allows for when the proportion of the images is different.

File: source/_footer.erb
Comment (MikeRogers0) [reply source/_footer.erb:1]: From what I understanding styling on the tag name isn't recommended, neither is the ID.

My intension is to create another PR where I switch the styling to use an approach like `.footer`/`.footer__block--modifier` ;)

File: lib/lrug_helpers.rb
Comment (MikeRogers0) [reply lib/lrug_helpers.rb:167]: Hah, this method was kind of funny to stumble upon. I'm sure at some point I'll update this in it's own PR :)

File: source/meetings/_meeting.erb
Comment (MikeRogers0) [reply source/meetings/_meeting.erb]: Updated @h-lame :)

File: source/_footer.erb
Comment (h-lame) [reply source/_footer.erb:1]: Let's not use BEM, it's gross.  Seems weird that "best practice" would push us to do `<footer class="footer">` as the `class` is clearly redundant.  But 🤷.

Comment (MikeRogers0) [reply source/_footer.erb:1]: Oh no! I love BEM, but I can compromise on how we name the classes :)

Though the main reason I really like using classes for styling, is it'll mean we have a consistent level of [CSS specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity), which'll be less surprises to anymore else wanting to jump on board :)

Comment (h-lame) [reply source/_footer.erb:1]: Hmm, it's not really a hill I want to die on, and you're doing the work so it's churlish of me to impose my antiquated reckons on it.  It just seems weird to me is all, and I am certainly not a frontend specialist these days so someone closer to things should probably make the call.  Might be willing to die on the anti-BEM hill though 😉

File: source/stylesheets/components/_sponsors.scss
Comment (h-lame) [reply source/stylesheets/components/_sponsors.scss:23]: I think it's fine in this PR and if it fixes something that's cool.  I guess the new grid layout is ironically less flexible than the old "float a div" - suspect the old div just expanded to what it contained, or we'd fixed it at a given size.  When adding sponsor logos for the sidebar I make sure they're all the same width (give or take a few px probably), but the ratios are all over because of that.

Thanks for explaining though!  Looks good.

**Review by h-lame — APPROVED** [review]

**Comment by MikeRogers0** [comment]
image

@h-lame Not a bad improvement 🥇

**Pull request merged**
